src/compiler/scala/tools/nsc/typechecker/RefChecks.scala

```diff
@@ -703,8 +720,13 @@ abstract class RefChecks extends Transform {
                           // TODO: what is the optimal way to test for a raw type at this point?
                           // Compilation has already failed so we shouldn't have to worry overmuch
                           // about forcing types.
-                          if (underlying.isJavaDefined && pa.typeArgs.isEmpty && abstractSym.typeParams.nonEmpty)
-                            ". To implement a raw type, use %s[_]".format(pa)
+                          if (underlying.isJavaDefined && pa.typeArgs.isEmpty && abstractSym.typeParams.nonEmpty) {
+                            prepareRawTypeForMessage(abstractSym).fold(
+                              ". To implement a raw type, use %s[_]".format(pa)
```

Comment:
Yes I will look into this in a week or so, when I have access to a real computer. By 'make this work for the whole compiler' do you mean resolving all subsequent test failures?